unified-runtime/source/adapters/opencl/adapter.cpp

```diff
@@ -90,21 +88,16 @@ urAdapterGet(uint32_t NumEntries, ur_adapter_handle_t *phAdapters,
   return UR_RESULT_SUCCESS;
 }
 
-UR_APIEXPORT ur_result_t UR_APICALL urAdapterRetain(ur_adapter_handle_t) {
-  ++adapter->RefCount;
+UR_APIEXPORT ur_result_t UR_APICALL
+urAdapterRetain(ur_adapter_handle_t hAdapter) {
+  ++hAdapter->RefCount;
   return UR_RESULT_SUCCESS;
 }
 
-UR_APIEXPORT ur_result_t UR_APICALL urAdapterRelease(ur_adapter_handle_t) {
-  // Check first if the adapter is valid pointer
-  if (adapter) {
-    std::lock_guard<std::mutex> Lock{adapter->Mutex};
-    if (--adapter->RefCount == 0) {
-      if (cl_ext::ExtFuncPtrCache) {
-        delete cl_ext::ExtFuncPtrCache;
-        cl_ext::ExtFuncPtrCache = nullptr;
-      }
-    }
+UR_APIEXPORT ur_result_t UR_APICALL
+urAdapterRelease(ur_adapter_handle_t hAdapter) {
+  if (--hAdapter->RefCount == 0) {
```

Comment:
Recently found out this is semantically distinct from `hAdapter->RefCount--`. Not a comment I'm just still mad about it.